src/jdk.crypto.ec/share/native/libsunec/impl/mpi.c

```diff
@@ -1727,12 +1727,15 @@ int    mp_cmp_mag(mp_int *a, mp_int *b)
  */
 int    mp_cmp_int(const mp_int *a, long z, int kmflag)
 {
+  mp_err  res;
   mp_int  tmp;
   int     out;
 
   ARGCHK(a != NULL, MP_EQ);
 
-  mp_init(&tmp, kmflag); mp_set_int(&tmp, z);
+  if((res = mp_init(&tmp, kmflag)) != MP_OKAY)
+    return res;
```

Comment:
`mp_err` is not the right type to return here. We could return `MP_EQ` in case of trouble (see line 1734), but that isn't strictly correct either.

The only use I could find is with `z == -3`, so something similar to the implementation of `mp_cmp_d()` would work here (a single `mp_digit` can hold the value `3`).

Or we could keep it simple:
```c
  if(mp_init(&tmp, kmflag) != MP_OKAY)
    return MP_EQ;
```
I'm not sure how much effort we should expend here.

Reply:
I rewrote the code to avoid the allocation.